Sources/Core/Microsoft.StreamProcessing/Operators/Afa/CompiledPartitionedAfaPipe_SingleEvent.cs

```diff
@@ -274,21 +273,19 @@ public override unsafe void OnNext(StreamMessage<TKey, TPayload> batch)
                                             {
                                                 if (this.isFinal[ns])
                                                 {
-                                                    var otherTime = Math.Min(synctime + this.MaxDuration, StreamEvent.InfinitySyncTime);
-
                                                     if (!this.IsSyncTimeSimultaneityFree)
                                                     {
                                                         var tentativeOutputEntry = this.tentativeOutput.entries[partitionIndex].value;
 
                                                         int ind = tentativeOutputEntry.Insert(src_hash[i]);
-                                                        tentativeOutputEntry.Values[ind].other = otherTime;
+                                                        tentativeOutputEntry.Values[ind].other = synctime + this.MaxDuration;
```

Comment:
Can `synctime + MaxDuration` overshoot MaxDate ?

Reply:
This undoes the fix done in commit 5d2dfe78

Reply:
Not really, MaxDuration is up to 7 days, synctime is the time now on timestamp. If we are close to year 9998 yes but that is not the case :)

Reply:
That is a good question, should we continue using the fix on 5d2dfe7 (https://github.com/microsoft/Trill/commit/5d2dfe7815ac883d3ddeea79091a6df50ec6443e). @peterfreiling

Reply:
Why would we want to revert 5d2dfe7? Please keep 5d2dfe7 .

---
In reply to: [493041257](https://github.com/microsoft/Trill/pull/134#discussion_r493041257) [](ancestors = 493041257)